<launch>
    <!-- 首先启动 grounding_sam_ros 的 launch 文件 -->
    <include file="$(find grounding_sam_ros)/launch/gsam.launch"/>
    
    <!-- 获取环境变量 -->
    <arg name="semantic_map_venv" default="/usr/bin/python" doc="3D动态语义地图generator, manager, guide的Python环境路径"/>
    <arg name="semantic_categories_json_path" default="$(env YANBOT_WS)/src/Cerebellum/semantic_map_pkg/scripts/semantic_categories.json"/>

    <!-- generator节点 -->
    <node name="semantic_map_generator_node" 
        pkg="semantic_map_pkg" 
        type="semantic_map_generator_node.py"
        launch-prefix="$(arg semantic_map_venv)"
        output="screen"
        respawn="true">
        <param name="semantic_categories_json_path" 
                type="str" 
                value="$(arg semantic_categories_json_path)"/>
        <param name="downsample_step"      type="int"   value="2" />
        <param name="topic_image_sub"      type="str"   value="/camera/color/image_raw" />
        <param name="topic_depth_sub"      type="str"   value="/camera/aligned_depth_to_color/image_raw" />
        <param name="topic_camera_info_sub" type="str" value="/camera/color/camera_info" />
        <param name="service_semantic_object_detseg" type="str" value="/vit_detection" />
        <param name="enable_filtering"    type="bool"  value="true" />
        <param name="topic_annotated_pub"  type="str"   value="/semantic_annotated" />
        <param name="topic_masks_pub"     type="str"   value="/semantic_masks" />
        <param name="service_update_prompt" type="str"  value="/update_prompt" />
        <param name="timer_duration"      type="double" value="1.0" />
    </node>
            



</launch>
